src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2503,6 +2503,26 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         return flags;
     }
 
+    til::point TermControl::_toControlOrigin(const til::point& terminalPos)
+    {
+        const til::size fontSize{ til::math::flooring, CharacterDimensions() };

```

Comment:
This results in rounding bugs at fractional display scalings, because `CharacterDimensions()` is in DIPs (at least according to its description).

For code based in DIPs I'd avoid using `til::size` and `til::point` entirely and just do the math with 2 regular floats (since the `til` structs are integers). And at the final step, I wouldn't use `flooring` but rather `rounding`, because `flooring` rounds down even if its `.9999999987f` or something.